api/python/ai/chronon/cli/tui/widgets.py

```
@@ -0,0 +1,573 @@
+        
+        def _find_blocking_recursive(node: str, step: StepStatus):
+            if step in visited:
+                return
+            visited.add(step)
+            
+            # Check dependencies - overlap check handles exact matches too
+            for dep in step.step_dependencies:
+                if dep.node_name in self.node_statuses:
+                    for other_step in self.node_statuses[dep.node_name]:
+                        if (other_step.status != "finished" and 
+                            self._steps_overlap(other_step, dep.start_date, dep.end_date)):
```

Flake8 (7.2.0)

[error] visually indented line with same indent as next logical line

(E129)